mkosi

```diff
@@ -3032,6 +3032,101 @@ class BooleanAction(argparse.Action):
+
+    Command line arguments starting with - or --are considered as regular arguments. Arguments
+    starting with @ are considered as files which are fed to the ini file parser implemented
+    in this class.
+    """
+
+    # Mapping of parameters supported in config files but not as command line arguments.
+    SPECIAL_MKOSI_DEFAULT_PARAMS = {
+        'QCow2': '--qcow2',
+        'OutputDirectory': '--output-dir',
+        'XZ': '--xz',
+        'NSpawnSettings': '--settings',
+        'ESPSize': '--esp-size',
+        'CheckSum': '--checksum',
+        'BMap': '--bmap',
+        'Packages': '--package',
+        'ExtraTrees': '--extra-tree',
+        'SkeletonTrees': '--skeleton-tree',
+        'BuildPackages': '--build-package',
+        'PostInstallationScript': '--postinst-script',
+    }
+
+    fromfile_prefix_chars = '@'
+
+    def __init__(self, *kargs, **kwargs):
+        self._ini_file_section = ""
+        self._ini_file_key = ""  # multi line list processing
+        self._ini_file_list_mode = False
+
+        # Add config files to be parsed
+        kwargs['fromfile_prefix_chars'] = __class__.fromfile_prefix_chars
+        super().__init__(*kargs, **kwargs)
+
+    def _read_args_from_files(self, arg_strings):
+        """Convert @ prefixed command line arguments with corresponding file content
+
+        Regular arguments are just returned. Arguments prefixed with @ are considered as
+        configuration file paths. The settings of each file are parsed and returned as
+        command line arguments.
+        Example:
+          The following mkosi.default is loaded.
+          [Distribution]
+          Distribution=fedora
+
+          mkosi is called like: mkosi -p httpd
+
+          arg_strings: ['@mkosi.default', '-p', 'httpd']
+          return value: ['--distribution', 'fedora', '-p', 'httpd']
+        """
+        def camel_to_arg(camel):
+            s1 = re.sub('(.)([A-Z][a-z]+)', r'\1-\2', camel)
+            return re.sub('([a-z0-9])([A-Z])', r'\1-\2', s1).lower()
+
+        def ini_key_to_cli_arg(key):
+            try:
+                return __class__.SPECIAL_MKOSI_DEFAULT_PARAMS[key]
+            except KeyError:
+                return '--' + camel_to_arg(key)
+
+        # expand arguments referencing files
+        new_arg_strings = []
+        for arg_string in arg_strings:
+            # for regular arguments, just add them back into the list
+            if not arg_string or arg_string[0] not in self.fromfile_prefix_chars:
+                new_arg_strings.append(arg_string)
+            # replace arguments referencing files with the file content
+            else:
+                try:
+                    with open(arg_string[1:]) as args_file:
+                        config = configparser.ConfigParser(delimiters='=')
+                        config.optionxform = str  # type: ignore
+                        config.read_file(args_file)
```

Comment:
the `with` context should be only arround this `read_file`. No need to keep the file open afterwards.